diff -ru emacs-29.4.orig/src/treesit.c emacs-29.4/src/treesit.c
--- emacs-29.4.orig/src/treesit.c	2024-06-21 22:10:07.000000000 +0900
+++ emacs-29.4/src/treesit.c	2026-04-08 11:26:16.283417000 +0900
@@ -3628,7 +3628,8 @@
 Emacs first looks in the directories mentioned in this variable,
 then in the `tree-sitter' subdirectory of `user-emacs-directory', and
 then in the system default locations for dynamic libraries, in that order.  */);
-  Vtreesit_extra_load_path = Qnil;
+#define BUILTIN_LANG_DIR "/Applications/Emacs.app/Contents/MacOS/tree-sitter"
+  Vtreesit_extra_load_path = list1 (build_pure_c_string (BUILTIN_LANG_DIR));
 
   staticpro (&Vtreesit_str_libtree_sitter);
   Vtreesit_str_libtree_sitter = build_pure_c_string ("libtree-sitter-");
